src/frontend/planner_test/tests/testdata/input/short_circuit.yaml

```diff
@@ -40,6 +40,12 @@
   expected_outputs:
   - logical_plan
   - batch_plan
+- id : short_circuit_optimize_time
+  sql: |
+    select true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true and true;
+  expected_outputs:
+    - logical_plan
+    - batch_plan
```

Comment:
maybe we can add an explict timeout parameter for the planner test?

Reply:
I have tested it locally, but do we have a timeout parameter available in the planner test right now?

Reply:
Since this is an improvement on the complexity, I suppose the query will result in unreasonable long processing time and times out the tests on CI if there's any regression. So it seems good enough now.